src/toil/serviceManager.py

```diff
@@ -276,7 +280,12 @@ def _startServices(jobDescriptionsWithServicesToStart,
                 for jobDesc in jobDescriptionsToRemove:
                     del servicesRemainingToStartForJob[jobDesc]
 
-def blockUntilServiceGroupIsStarted(jobDesc, jobDescriptionsWithServicesThatHaveStarted, jobDescriptionsWithServicesThatHaveFailedToStart, serviceJobsToStart, terminate, jobStore):
+def blockUntilServiceGroupIsStarted(jobDesc: ServiceJobDescription,
+                                    jobDescriptionsWithServicesThatHaveStarted: 'Queue[ServiceJobDescription]',
+                                    jobDescriptionsWithServicesThatHaveFailedToStart: 'Queue[ServiceJobDescription]',
```

Comment:
It looks like `jobDescriptionsWithServicesThatHaveFailedToStart` isn't used within the function, should we remove it as a param?

Reply:
Yeah, I think we used to fail services here, and I added that long comment that this would make deadlocks happen and changed the behavior but not the interface. So we should probably remove it. You can add that to the PR if you want.